### ISSUE TITLE
A csproj project w/o any local .cs files can fail to load .editorconfig

### ISSUE THREAD

**Issue by yaakov-h** [dotnet/roslyn#67660]
### Describe the bug
When a SDK C# project contains no local C# files, but Directory.Build.targets includes a C# file via `<Compile Include=.... />`, the `.editorconfig` file is not found and can trigger analyzer false-positives.

### To Reproduce

1. Clone https://github.com/yaakov-h/dotnet-sdk-repro-nocs
2. Run `dotnet clean && dotnet build`.
3. Note that there are no errors or warnings produced.
4. Uncomment the `<Compile>` element in `Directory.Build.targets`
5. Run `dotnet clean && dotnet build`.
6. Note that there are now two warnings produced, SA0001 and SA1633, both of which are disabled in `.editorconfig`.

This is where things get weird.

7. Rename `Directory.Build.targets` to `Directory.Build.props`
8. Run `dotnet clean && dotnet build`.
9. Note that there are no errors or warnings produced.

Additionally:

10. Rename `Directory.Build.props` back to `Directory.Build.targets`, to undo the side-effects of the above test.
11. Create a new file, `nocs\FileName.cs` which contains valid C#. I used the autogenerated VS template below.
12. Run `dotnet clean && dotnet build`.
13. Note that there are no errors or warnings produced.

### Additional Code

FileName.cs, for step 11 above:

```csharp
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace nocs
{
    internal class FileName
    {
    }
}
```
### Further technical details

Visual Studio 2022 Professional Version 17.5.1

dotnet:

```text
.NET SDK:
 Version:   7.0.201
 Commit:    68f2d7e7a3

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.201\

Host:
  Version:      7.0.3
  Architecture: x64
  Commit:       0a2bda10e8

.NET SDKs installed:
  6.0.310 [C:\Program Files\dotnet\sdk]
  6.0.407 [C:\Program Files\dotnet\sdk]
  7.0.201 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.14 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**Comment by jcouv** [comment]
When the editor config discovery is not sufficient for a scenario, the fallback is to specify the editor config files (see below). I don't think we're going to change editor config discovery to accommodate such edge case scenarios.

```
   <PropertyGroup>
     <TargetFramework>net70</TargetFramework>
     <OutputType>exe</OutputType>
     <DiscoverEditorConfigFiles>false</DiscoverEditorConfigFiles>
   </PropertyGroup>
   <ItemGroup>
    <EditorConfigFiles Include=".editorconfig" />
   </ItemGroup>
```
